base/src/ipfixcol.c

```diff
@@ -255,27 +255,39 @@ int main (int argc, char* argv[])
 		MSG_SYSLOG_INIT(PACKAGE);
 		
 		/* and send all following messages to the syslog */
-		if (daemon (1, 0)) {
+		if (daemon(1, 0)) {
 			MSG_ERROR(msg_module, "%s", strerror(errno));
 		}
 	}
 
-	/* check config file */
-	if (config_file == NULL) {
+	/* Check config file */
+	if (startup_config == NULL) {
 		/* and use default if not specified */
-		config_file = DEFAULT_CONFIG_FILE;
-		MSG_NOTICE(msg_module, "Using default configuration file: %s", config_file);
+		startup_config = DEFAULT_STARTUP_CONFIG;
+		MSG_NOTICE(msg_module, "Using default configuration file: %s", startup_config);
+	} else if (access(startup_config, R_OK) == -1) {
```

Comment:
Another problem introduce here is that you do the check only for user given files, but the defaults might be missing as well. I understand the problem you are trying to solve and I think that a plausible solution is the following:
- startup_config: Do nothing. It is already handled in config_open_xml.
- internal_config: Extract ic_init call from get_*_plugins in config.c to special static function (such as get_xml_context) and handle the inaccessible file there. Since it is called upon initialization, you'll get the error immediately upon startup.
- ipfix_elements: This is the biggest problem since storage plugin initialization is waiting for first data, as explained in #125. There are three options:
1) fix #125 and open the ipfix_element during storage plugin initialization. Disadvantage is that the check is performed by the the plugin, which might get it wrong.
2) Open the file in collector and pass a) the content b) file descriptor
3) Just do the check here as you are doing it now, explain in comment why the check is here. It is also possible to use config_open_xml and then close and discard it immediately. 
4) Let the collector provide an API for working with the ipfix_elements file. This would allow to open/check the file once in the collector and the plugins would not have to worry about it. It'll make the plugin code a bit easier and it would also allow to support reconfiguration of this file in the future.

I do not like 1) since it relies on plugins to do the check. The 2) means complicating the API, 3) introduces time of check, time of use race condition, so the best option in my opinion is 4). I think that fixing #125 is the first step which will give you what you need and the 4) option will solve this for good. What do you think?